packages/components/src/navigation/menu/use-navigation-tree-menu.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useNavigationContext } from '../context';
+import { ROOT_MENU } from '../constants';
+
+export const useNavigationTreeMenu = ( props ) => {
+	const {
+		navigationTree: { addMenu, removeMenu },
+	} = useNavigationContext();
+
+	const key = props.menu || ROOT_MENU;
+	useEffect( () => {
+		addMenu( key, { ...props } );
```

Comment:
Not a big deal, but just for the sake of consistency...

Root menus that don't specify the `menu` prop would end up with an empty `menu` property here.

<img width="665" alt="Screenshot 2020-09-15 at 16 35 03" src="https://user-images.githubusercontent.com/2070010/93232123-6ec52380-f771-11ea-85d0-b928e8d95632.png">

This would be quickly solvable with:
```js
addMenu( key, { ...props, menu: key } );
```

Reply:
Nice find! Done!